### PR TITLE
Stripes to csp layout

### PR DESCRIPTION
**Pull request by ZijiaZhang**
Please Test using strips_elevator.   about #142 

A quick preview:
image

### CODE REVIEW

File: aipython/cspProblem.py [hunk 1]
```diff
@@ -65,7 +65,11 @@ def __init__(self, domains, constraints, positions={}):
             if occurence > 1:
                 for j in range(occurence):
                     # start numbering
+                    old_repr = self.constraints[i + j].repr
+                    old_posx,old_posy = self.positions[old_repr]
+                    gap = min(200/occurence,50)
                     self.constraints[i + j].repr += str(j)
+                    self.positions[self.constraints[i + j].repr] = (old_posx,old_posy+gap*j)
```
Comment (Chenliang-Zhou) [inline aipython/cspProblem.py]: comment in the code why you used `gap`

File: aipython/stripsCSPPlanner.py [hunk 2]
```diff
@@ -25,42 +25,84 @@ def __init__(self, planning_problem, horizon=2):
         prob_domain = planning_problem.prob_domain
         initial_state = planning_problem.initial_state
         goal = planning_problem.goal
+
+        old_pos = planning_problem.positions
+
+        maxwidth = max([x for x, y in old_pos.values()])
```
Comment (Chenliang-Zhou) [inline aipython/stripsCSPPlanner.py]: It seems that you need to deal with the empty array


image

**Comment by Chenliang-Zhou** [comment]
The first TODO is not done yet:

Change each constraint's name to be unique and **representative** (see AIspace). You can change a constraint name by constraint.repr = the_name_you_want.

For examples, start_elevator, end_passenger, pre_action0, effect_action0, etc.

File: aipython/stripsCSPPlanner.py [hunk 3]
```diff
@@ -59,9 +77,73 @@ def __init__(self, planning_problem, horizon=2):
...
+        while i < len(constraint_strings):
+            occurence = constraint_strings.count(constraint_strings[i])
+            if occurence > 1:
+                old_repr = constraints[i].repr
+                if old_repr in positions:
+                    old_posx, old_posy = positions[old_repr]
+                    
+                for j in range(occurence):
+                    # start numbering
+                    gap = min(200 / occurence, 50)
```
Comment (Chenliang-Zhou) [inline aipython/stripsCSPPlanner.py]: please write a comment here as it is hard to understand what is `gap` for.

Comment (ZijiaZhang) [reply aipython/stripsCSPPlanner.py]: I have Changed the name of the constraint. Therefore, it is no longer needed to handle the duplicated names, and I have removed these code.

**Pull request merged**